lib/router/index.js

```diff
@@ -59,6 +61,24 @@ function throwIfNoRouter () {
 // Export the SingletonRouter and this is the public API.
 export default SingletonRouter
 
+export function withRouter (options) {
+  return function withRouterComponent (Comp) {
+    const compName = Comp.displayName || Comp.name || 'Component'
+
+    return class ComponentWithRouter extends Component {
```

Comment:
FYI [this is](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/modules/withRouter.js) the standard way it looks like